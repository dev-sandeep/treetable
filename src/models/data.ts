export interface DataObjTs{
    data:any,
    kids:any    
}