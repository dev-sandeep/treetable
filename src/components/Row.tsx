/**
 * responsible for rendering a single row of any given table
 * @author Sandeep G
 * @since 20201001
 */
import React from 'react';
import {TEMP_REF}  from './../common/constants'
import {RowComponentTs} from './../models/table'
import styled from 'styled-components'
import {ThWrapper} from './../components/TreeTable'

const TdWrapper = styled.td`
    padding: 10px;
    border: 0.1px solid gray;
    width: auto;
`;


const Row = ({ data, type }:RowComponentTs) => {
       
    return (
        <>
            {Object.keys(data).map((key, index) => (
                //in case TEMP_REF hide the cell
                <React.Fragment key={key+index}>
                    {type === "header" ? 
                        <ThWrapper className={key === TEMP_REF?'hidden':''}>{key}</ThWrapper> : 
                            <TdWrapper className={key === TEMP_REF?'hidden':''}>{data[key]}</TdWrapper>}
                </React.Fragment>
            ))}
        </>
    );
}

export default Row;
