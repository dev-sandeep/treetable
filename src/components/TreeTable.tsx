import React from 'react';
import { DataObjTs } from './../models/data'
import { TableTs } from './../models/table'
import Row from './Row'
import styled from 'styled-components'
import TableBody from './TableBody'

const StyledTableWrapper = styled.table.attrs(() => ({ cellSpacing: "0" }))`
    width: 100%;
    padding: 10px;
`;

export const TrWrapper = styled.tr`
    border: 0.1px solid gray;
    width: auto;
    padding: 10px;
`;

export const ThWrapper = styled.th`
    border: 0.1px solid gray;
    width: auto;
    padding: 10px;
    background: gainsboro;
`;

const TableHeaderH3 = styled.h3`
    padding-left: 10px;
    margin-bottom: 0px;
`;

/**
 * responsible for loading the header row
 */
interface TableHeaderTs {
    data: DataObjTs[];
}

const TableHeader = ({ data }: TableHeaderTs) => {
    if (data.length === 0)
        return null;

    const headerData = data[0];
    return (<thead>
        <TrWrapper>
            {/* empty header for expand/collapse btn */}
            <ThWrapper></ThWrapper>
            <Row type="header" data={headerData.data} />
            {/* empty header for delete btn */}
            <ThWrapper></ThWrapper>
        </TrWrapper>
    </thead>)
}

const NoData = () => (
    <>
        <h2>No Data- :(</h2>
        <h5>You must have deleted everything, refresh the page!</h5>
    </>
);

const TreeTable = ({ queryData, title }: TableTs):JSX.Element => {
    return (
        <>
            {queryData.length === 0 ? (<NoData />) : (
                <>
                    <TableHeaderH3>{title || 'Main Table'}</TableHeaderH3>
                    <StyledTableWrapper>
                        <TableHeader data={queryData} />
                        <TableBody data={queryData} />
                    </StyledTableWrapper>
                </>
            )}
        </>
    );
}

export default TreeTable;