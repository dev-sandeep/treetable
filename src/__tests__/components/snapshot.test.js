import React from 'react'
import renderer from 'react-test-renderer'
import TreeTable from '../../components/TreeTable'
import Row from '../../components/Row'
import {jsonData} from './../../data'

it('snapshot test for table-body when no data', () => {
const tree = renderer
    .create(<TreeTable queryData={[]} title="Main table" />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshot test for Row', () => {
const tree = renderer
    .create(<Row data={jsonData[0].data} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});