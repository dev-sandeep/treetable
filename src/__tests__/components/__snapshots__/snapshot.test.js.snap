// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Row 1`] = `
Array [
  <td
    className="sc-bdnylx enjUwz"
  >
    34
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    Joqmo
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    female
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    BITES
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    6.2000000000
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    98
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    Mon Dec 13 00:00:00 CET 1993
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    Wed Dec 24 07:14:50 CET 2014
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    67035
  </td>,
  <td
    className="sc-bdnylx enjUwz"
  >
    true
  </td>,
]
`;

exports[`snapshot test for table-body when no data 1`] = `
Array [
  <h2>
    No Data- :(
  </h2>,
  <h5>
    You must have deleted everything, refresh the page!
  </h5>,
]
`;

exports[`snapshot test for tablebody 1`] = `
Array [
  <h2>
    No Data- :(
  </h2>,
  <h5>
    You must have deleted everything, refresh the page!
  </h5>,
]
`;
